django/db/backends/sqlite3/creation.py

```diff
@@ -49,18 +51,28 @@ def _create_test_db(self, verbosity, autoclobber, keepdb=False):
         return test_database_name
 
     def get_test_db_clone_settings(self, suffix):
-        orig_settings_dict = self.connection.settings_dict
-        source_database_name = orig_settings_dict["NAME"]
-        if self.is_in_memory_db(source_database_name):
-            return orig_settings_dict
-        else:
-            root, ext = os.path.splitext(orig_settings_dict["NAME"])
-            return {**orig_settings_dict, "NAME": "{}_{}{}".format(root, suffix, ext)}
+        settings_dict = self.connection.settings_dict
+
+        source_database_name = settings_dict["NAME"]
+        if not self.is_in_memory_db(source_database_name):
+            root, ext = os.path.splitext(source_database_name)
+            return {**settings_dict, "NAME": f"{root}_{suffix}{ext}"}
+
+        start_method = multiprocessing.get_start_method()
+        if start_method == "fork":
+            return settings_dict
+        if start_method == "spawn":
+            return {
+                **settings_dict,
+                "NAME": f"{self.connection.alias}_{suffix}.sqlite3",
```

Comment:
`...{suffix}{ext}` was used above, but here we're hard-coding the extension `...{suffix}.sqlite3`. I'm _presuming_ that's an accidental divergence, but it could also be e.g. a specific requirement for spawn to work on Windows or something ...?

Reply:
Thanks for the comment. 👍 

When running it locally on Windows at this point (for me) the database is in-memory so the string returned by `os.path.splittext` is `file:memorydb_default?mode=memory&cache=shared'`. We therefore can't use the `root`/`ext`  logic as above as we want to be migrating the in-memory database to disk with spawn. 

(That's how I understand it anyway)

Reply:
Ah, that makes a lot of sense.  Thanks :)